client/src/Components/AddVideo.js

```diff
@@ -0,0 +1,62 @@
+import { useState } from "react";
+
+const AddVideo = ({ videos, setVideos }) => {
+  const [videoName, setVideoName] = useState("");
+  const [videoLink, setVideoLink] = useState("");
+  function handleInputChange(event) {
+    if (event.target.name === "name") {
+      setVideoName(event.target.value);
+    } else if (event.target.name === "url") {
+      setVideoLink(event.target.value);
+    }
+  }
+
+  function handleSubmit(event) {
```

Comment:
When you log videos after updating it, you may notice that it doesn't immediately reflect the new state because `setState` is `asynchronous`. If you want to see the updated state, you can log it inside a useEffect hook with videos as a dependency. Might worth trying and compare it with useState 😃